feat(sbb-toggle-check): initial implementation (#1283)
lyne-design-system/lyne-components: src/components/sbb-toggle-check/readme.md

`<sbb-toggle-check>` provides the same functionality as a native `<input type="checkbox">`
enhanced with the SBB Design.

```html
<sbb-toggle-check
  value="single-checkbox"
  disabled="false"
  checked="false"
>
  Example
</sbb-toggle-check>
```

## Label

The label is provided as the content to the `<sbb-toggle-check>` element.
It can be displayed after or before the toggle-check.

If you don't want the label to appear next to the toggle-check, you can use

> [`aria-label`](https://www.w3.org/TR/wai-aria/states_and_properties#aria-label) or

Reviewer: This is incorrect, as we use the properties `accessibility-*` to place the actual `aria-*` attributes on the checkbox.

[`aria-labelledby`](https://www.w3.org/TR/wai-aria/states_and_properties#aria-labelledby) to
specify an appropriate label.

## Accessibility

`SbbToggleCheck` uses an internal `<input type="checkbox">` to provide an accessible experience.
This internal checkbox receives focus and is automatically labelled by the text content of the
`<sbb-toggle-check>` element. Avoid adding other interactive controls into the content of
`<sbb-toggle-check>`, as this degrades the experience for users of assistive technology.

> Always provide an accessible label via `aria-label` or `aria-labelledby` for checkboxes without

Reviewer: See above comment.

descriptive text content. For dynamic labels, `SbbToggleCheck` provides input properties for binding
`aria-label`, `aria-labelledby` and `aria-labelledby`. This means that you should not use the `attr.` prefix when
binding these properties, as demonstrated below.

```html

> <sbb-toggle-check aria-label="isSubscribedToEmailsMessage" />

Reviewer: See above comment.

`<sbb-toggle-check>` provides the same functionality as a native `<input type="checkbox">`
enhanced with the SBB Design.

```html
<sbb-toggle-check
  value="single-checkbox"
  disabled="false"
  checked="false"
>
  Example
</sbb-toggle-check>
```

## Label

The label is provided as the content to the `<sbb-toggle-check>` element.
It can be displayed after or before the toggle-check.

If you don't want the label to appear next to the toggle-check, you can use
[`accessibility-label`](https://www.w3.org/TR/WCAG20-TECHS/ARIA14.html) or
[`accessibility-labelledby`](https://www.w3.org/TR/WCAG20-TECHS/ARIA16.html) to
specify an appropriate label.

## Accessibility

`SbbToggleCheck` uses an internal `<input type="checkbox">` to provide an accessible experience.
This internal checkbox receives focus and is automatically labelled by the text content of the
`<sbb-toggle-check>` element. Avoid adding other interactive controls into the content of
`<sbb-toggle-check>`, as this degrades the experience for users of assistive technology.

> Always provide an accessible label via `accessibility-label` or `accessibility-labelledby` for checkboxes without
> descriptive text content. For dynamic labels, `SbbToggleCheck` provides input properties for binding
> `aria-label` and `aria-labelledby`. This means that you should not use the `attr.` prefix when
> binding these properties, as demonstrated below.
> 
> Additionaly you can use [`accessibility-describedby`](https://www.w3.org/TR/WCAG20-TECHS/ARIA1.html) to describe the element on which the attribute is set.

Reviewer: ```suggestion
Always provide an accessible label via `accessibility-label` or `accessibility-labelledby` for checkboxes without
descriptive text content.

Additionaly you can use `accessibility-describedby` (forwarded as [`aria-describedby`](https://www.w3.org/TR/WCAG20-TECHS/ARIA1.html))
to describe the element on which the attribute is set.
```

```html

> <sbb-toggle-check aria-label="isSubscribedToEmailsMessage" />

Reviewer: ```suggestion
<sbb-toggle-check accessibility-label="Subscribed to email message" />
```
